cluster.go

```diff
@@ -189,6 +189,15 @@ func (c *Cluster) Validate(isTmpl bool) error {
 		}
 	}
 
+	nodeAddressSet := make(map[string]int)
```

Comment:
For the best performance, it is advised to use `map[string]struct{}`.

For the best convenience, it is good to use `map[string]bool` because it can be used as:

```go
    if nodeAddressSet[n.Address] {
        return err
    }
```